Can't perform make static-release on master HEAD

I can't build static nydus binaries on the master HEAD. Should I add more dependencies or extra configurations
?

```
gechangwei@n227-089-202 ~/git_repo/nydus-rs ((HEAD detached at origin/master))$ make static-release
/home/gechangwei/.cargo/bin/cargo fmt -- --check
/home/gechangwei/.cargo/bin/cargo build  --features=virtiofs --target x86_64-unknown-linux-musl --release
   Compiling libz-sys v1.1.8
   Compiling tokio-native-tls v0.3.0
   Compiling nydus-app v0.3.0 (/data00/home/gechangwei/git_repo/nydus-rs/app)
   Compiling spmc v0.3.0
   Compiling rust-fsm v0.6.0
   Compiling hyperlocal v0.8.0
   Compiling clap v4.0.18
   Compiling vhost-user-backend v0.7.0
   Compiling rlimit v0.9.0
   Compiling hyper-tls v0.5.0
error: failed to run custom build command for `libz-sys v1.1.8`

Caused by:
  process didn't exit successfully: `/data00/home/gechangwei/git_repo/nydus-rs/target/release/build/libz-sys-886804fd8bcbb43e/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LIBZ_SYS_STATIC
  cargo:rerun-if-changed=build.rs
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-musl = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_musl = None
  TARGET_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-musl = None
  CMAKE_GENERATOR_x86_64_unknown_linux_musl = None
  TARGET_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-musl = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_musl = None
  TARGET_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-unknown-linux-musl = None
  CMAKE_x86_64_unknown_linux_musl = None
  TARGET_CMAKE = None
  CMAKE = None
  running: "cmake" "/data00/home/gechangwei/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/libz-sys-1.1.8/src/zlib-ng" "-DBUILD_SHARED_LIBS=OFF" "-DZLIB_COMPAT=ON" "-DZLIB_ENABLE_TESTS=OFF" "-DWITH_GZFILEOP=ON" "-DCMAKE_SYSTEM_NAME=Linux" "-DCMAKE_SYSTEM_PROCESSOR=x86_64" "-DCMAKE_INSTALL_PREFIX=/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/musl-gcc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=musl-g++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/musl-gcc" "-DCMAKE_BUILD_TYPE=Release"
  -- Using CMake version 3.13.4
  -- ZLIB_HEADER_VERSION: 1.2.11
  -- ZLIBNG_HEADER_VERSION: 2.1.0.devel
  -- Arch not recognized, falling back to cmake arch: 'x86_64'
  -- Basearch of 'x86_64' has been detected as: 'x86'
  -- Architecture-specific source files: arch/x86/x86_features.c;arch/x86/slide_hash_avx2.c;arch/x86/chunkset_avx.c;arch/x86/compare256_avx2.c;arch/x86/adler32_avx2.c;arch/x86/chunkset_sse41.c;arch/x86/adler32_sse42.c;arch/x86/insert_string_sse42.c;arch/x86/chunkset_sse2.c;arch/x86/compare256_sse2.c;arch/x86/slide_hash_sse2.c;arch/x86/adler32_ssse3.c;arch/x86/crc32_fold_pclmulqdq.c
  -- The following features have been enabled:

   * CMAKE_BUILD_TYPE, Build type: Release (selected)
   * AVX2_SLIDEHASH, Support AVX2 optimized slide_hash, using "-mavx2"
   * AVX_CHUNKSET, Support AVX optimized chunkset, using "-mavx2"
   * AVX2_COMPARE256, Support AVX2 optimized compare256, using "-mavx2"
   * AVX2_ADLER32, Support AVX2-accelerated adler32, using "-mavx2"
   * SSE42_CRC, Support SSE4.2 optimized CRC hash generation, using "-msse4.2"
   * SSSE3_ADLER32, Support SSSE3-accelerated adler32, using "-mssse3"
   * PCLMUL_CRC, Support CRC hash generation using PCLMULQDQ, using "-mssse3 -msse4.2 -mpclmul"
   * WITH_GZFILEOP, Compile with support for gzFile related functions
   * ZLIB_COMPAT, Compile with zlib compatible API
   * WITH_SANITIZER, Enable sanitizer support
   * WITH_OPTIM, Build with optimisation
   * WITH_NEW_STRATEGIES, Use new strategies
   * WITH_AVX2, Build with AVX2
   * WITH_SSE2, Build with SSE2
   * WITH_SSSE3, Build with SSSE3
   * WITH_SSE41, Build with SSE41
   * WITH_SSE42, Build with SSE42
   * WITH_PCLMULQDQ, Build with PCLMULQDQ

  -- The following features have been disabled:

   * FORCE_TZCNT, Assume CPU is TZCNT capable
   * ZLIB_SYMBOL_PREFIX, Publicly exported symbols DO NOT have a custom prefix
   * ZLIB_ENABLE_TESTS, Build test binaries
   * ZLIB_DUAL_LINK, Dual link tests against system zlib
   * WITH_FUZZERS, Build test/fuzz
   * WITH_BENCHMARKS, Build test/benchmarks
   * WITH_BENCHMARK_APPS, Build application benchmarks
   * WITH_NATIVE_INSTRUCTIONS, Instruct the compiler to use the full instruction set on this host (gcc/clang -march=native)
   * WITH_MAINTAINER_WARNINGS, Build with project maintainer warnings
   * WITH_CODE_COVERAGE, Enable code coverage reporting
   * WITH_INFLATE_STRICT, Build with strict inflate distance checking
   * WITH_INFLATE_ALLOW_INVALID_DIST, Build with zero fill for inflate invalid distances
   * WITH_AVX512, Build with AVX512
   * WITH_AVX512VNNI, Build with AVX512 VNNI
   * WITH_VPCLMULQDQ, Build with VPCLMULQDQ
   * INSTALL_UTILS, Copy minigzip and minideflate during install

  -- Configuring done
  -- Generating done
  -- Build files have been written to: /data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build
  running: "cmake" "--build" "." "--target" "install" "--config" "Release"
  make[1]: Entering directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  make[2]: Entering directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  make[3]: Entering directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  make[3]: Leaving directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  make[3]: Entering directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  [  2%] Linking C static library libz.a
  make[3]: Leaving directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  make[2]: Leaving directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'
  make[1]: Leaving directory '/data00/home/gechangwei/git_repo/nydus-rs/target/x86_64-unknown-linux-musl/release/build/libz-sys-4f7840ca4eaea4d9/out/build'

  --- stderr
  Error running link command: No such file or directory
  make[3]: *** [CMakeFiles/zlib.dir/build.make:730: libz.a] Error 2
  make[2]: *** [CMakeFiles/Makefile2:73: CMakeFiles/zlib.dir/all] Error 2
  make[1]: *** [Makefile:118: all] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /home/gechangwei/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/cmake-0.1.49/src/lib.rs:1104:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
make: *** [Makefile:89: build] Error 101

```

--- a/Makefile
+++ b/Makefile
@@ -72,7 +72,7 @@ define build_golang
 	fi
 endef
 
-.PHONY: .release_version .format .musl_target \
+.PHONY: .release_version .format .musl_target .clean_libz_sys \
 	all all-build all-release all-static-release build release static-release
 
 .release_version:
@@ -84,6 +84,11 @@ endef
 .musl_target:
 	$(eval CARGO_BUILD_FLAGS += --target ${RUST_TARGET_STATIC})
 
+# Workaround to clean up stale cache for libz-sys
+.clean_libz_sys:
+	@${CARGO} clean --target ${RUST_TARGET_STATIC} -p libz-sys
+	@${CARGO} clean --target ${RUST_TARGET_STATIC} --release -p libz-sys
+
 # Targets that are exposed to developers and users.
 build: .format .release_version
 	${CARGO} build $(CARGO_COMMON) $(CARGO_BUILD_FLAGS)
@@ -92,7 +97,7 @@ build: .format .release_version
 
 release: .format .release_version build
 
-static-release: .musl_target .format .release_version build
+static-release: .clean_libz_sys .musl_target .format .release_version build
 
 clean:
 	${CARGO} clean

